### PR TITLE
config: validate instance/replicaset/group names

### PR DESCRIPTION
**Pull request by Totktonada**
Recently added persistent instance/replicaset/cluster names have certain validation rules (see #5029 and #9148). An instance name and a replicaset name that are provided in a declarative configuration are stored in the database, so they should follow the same rules.

This patch implements the validation for instance/replicaset/group names, for `--name` CLI option and `TT_INSTANCE_NAME` environment variable.

The validation rules are described in https://github.com/tarantool/doc/issues/3466, https://github.com/tarantool/doc/issues/3467 and https://github.com/tarantool/doc/issues/3468.

Part of #8862
Related to #9148

### CODE REVIEW

File: changelogs/unreleased/config-validate-names.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## bugfix/config
+
+* Validate instance, replicaset, group names in config, validate `--name` CLI
+  option and `TT_INSTANCE_NAME` environment variable (gh-8862).
```
Comment (p7nov) [inline changelogs/unreleased/config-validate-names.md:4]:
```suggestion
* Added the validation of instance, replica set, and group names in the
  configuration, `--name` CLI option, and `TT_INSTANCE_NAME` environment
  variable (gh-8862).
```

**Review by p7nov — APPROVED** [review]
Please see the changelog suggestion.

File: changelogs/unreleased/config-validate-names.md (hunk 1 above)
Comment (Totktonada) [reply changelogs/unreleased/config-validate-names.md]: Thanks! Changed.

**Review by Serpentian — APPROVED** [review]
Thank you for the patch! Looks great, I have no objections

**Review by ochaplashkin — APPROVED** [review]

**Pull request merged**